cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/main.py

```diff
@@ -0,0 +1,378 @@
+# Copyright 2018 Google LLC.
+# This software is provided as-is, without warranty or representation
+# for any use or purpose.
+# Your use of it is subject to your agreement with Google.
+
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Background Cloud Function for loading data to BigQuery.
+"""
+import json
+from collections import deque
+from os import getenv
+from pathlib import Path
+from typing import Any
+from typing import Dict, List, Optional, Tuple
+from urllib.parse import urlparse
+import logging
+from time import monotonic, sleep
+
+from google.api_core.client_info import ClientInfo
+from google.cloud import bigquery, storage
+from google.cloud.exceptions import NotFound
+
+# 15TB per BQ load job.
+MAX_BATCH_BYTES = 15 * 10 ** 12
+MAX_URIS_PER_LOAD = 10 ** 4
+
+BASE_LOAD_JOB_CONFIG = {
+    "sourceFormat": "CSV",
+    "fieldDelimiter": "|",
+    "writeDisposition": "WRITE_APPEND",
+}
+
+DEFAULT_BQ_LOAD_STATE_TABLE = "bigquery_loads.serverless_bq_loads"
+
+DEFAULT_EXTERNAL_TABLE_DEFINITION = {
+    "sourceFormat": "PARQUET"
+}
+
+DEFAULT_JOB_LABELS = {
+    "component": "event-based-gcs-ingest",
+    "cloud-function-name": getenv("FUNCTION_NAME"),
+}
+
+BASE_LOAD_JOB_CONFIG = {
+    "sourceFormat": "CSV",
+    "fieldDelimiter": "|",
+    "writeDisposition": "WRITE_APPEND",
+    "labels": DEFAULT_JOB_LABELS,
+}
+
+# Will wait up to this polling for errors before exiting
+# This is to check if job fail quickly, not to assert the succeed
+# This may not be honored if longer than cloud function timeout
+# https://cloud.google.com/functions/docs/concepts/exec#timeout
+WAIT_FOR_JOB_SECONDS = 5
+
+
+def main(event: Dict, context):
+    # Set by Cloud Function Execution Environment
+    # https://cloud.google.com/functions/docs/env-var
+    project = getenv("GCP_PROJECT")
+
+
+    bucket_id, object_id = parse_notification(event)
+
+    success_filename = getenv("SUCCESS_FILENAME", "_SUCCESS")
```

Comment:
Seems like this could be a variable defined above given `get_batches_for_prefix` depends on the same default